src/main/kotlin/org/rust/ide/newProject/ui/RsNewProjectPanel.kt

```diff
@@ -7,31 +7,38 @@ package org.rust.ide.newProject.ui
 
 import com.intellij.openapi.Disposable
 import com.intellij.openapi.options.ConfigurationException
-import com.intellij.ui.components.JBCheckBox
+import com.intellij.ui.components.JBRadioButton
 import org.rust.cargo.project.settings.ui.RustProjectSettingsPanel
 import org.rust.ide.newProject.ConfigurationData
 import org.rust.ide.ui.RsLayoutBuilder
 import java.awt.event.ItemListener
+import javax.swing.ButtonGroup
 
 class RsNewProjectPanel(
     private val showProjectTypeCheckbox: Boolean,
     updateListener: (() -> Unit)? = null
 ) : Disposable {
 
     private val rustProjectSettings = RustProjectSettingsPanel(updateListener = updateListener)
-    private val createBinaryCheckbox = JBCheckBox(null, true)
-    private val checkboxListener = ItemListener { updateListener?.invoke() }
+    private val templateButtons = listOf(JBRadioButton("Binary (application)", true), JBRadioButton("Library"))
+    private val checkboxListener = ItemListener {
+        if (it.stateChange == java.awt.event.ItemEvent.SELECTED) updateListener?.invoke()
```

Comment:
Hmm, why not import `java.awt.event.ItemEvent`?

Reply:
No reason, just failed 'shorten FQNs feature'